src/PlayerSkeleton.java

```diff
@@ -25,3 +26,204 @@ public static void main(String[] args) {
 	}
 	
 }
+

```

Comment:
One of the most important thing is that we might need to extract weights out. The Heuristic class just calculates the score for a specific state based on given weights. So as far as I am concern, the construction of Heuristic could be `Heuristics(State state, Double[] weights)` somethings like this

Reply:
actually the `Double[] weights` would come from a class called `Particles` similar to the code that I shared with you yesterday. The `Particles` class will store the weight vector and use PSO algorithm to update the weights by iterations.

Reply:
For now my idea is that we can keep this Heuristics function, but just use it as a "calculator". We can make it a static class or singleton, then with weights or, `Particle` and `State` provide we could return the corresponding score.